src/server/game/Spells/SpellEffects.cpp

```diff
@@ -1730,7 +1730,8 @@ void Spell::EffectEnergize(SpellEffIndex effIndex)
 
     Powers power = Powers(m_spellInfo->Effects[effIndex].MiscValue);
 
-    if (unitTarget->GetTypeId() == TYPEID_PLAYER && unitTarget->getPowerType() != power && !m_spellInfo->HasAttribute(SPELL_ATTR7_CAN_RESTORE_SECONDARY_POWER))
+    if (unitTarget->GetTypeId() == TYPEID_PLAYER && unitTarget->getPowerType() != power && !m_spellInfo->SpellFamilyName == SPELLFAMILY_POTION 
```

Comment:
`!m_spellInfo->SpellFamilyName == SPELLFAMILY_POTION` should be `m_spellInfo->SpellFamilyName != SPELLFAMILY_POTION`